src/main/docker/Dockerfile

```diff
@@ -42,35 +42,36 @@ RUN zypper -n refresh && \
     zypper -n clean --all
 
 #
-# Stage 2: Download and verify gosu
+# Stage 2: Download and compile su-exec
 #
 FROM stage1 AS stage2
 
-# Install dirmngr to enable gosu verification
-RUN zypper -n install dirmngr
-
-# Download and verify gosu
-RUN gpg --batch --keyserver-options http-proxy=${env.HTTP_PROXY} --keyserver hkps://keys.openpgp.org \
-        --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 && \
-    curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.17/gosu-amd64" && \
-    curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.17/gosu-amd64.asc" && \
-    gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu && \
-    chmod +x /usr/local/bin/gosu
+# Download and compile su-exec
+RUN zypper -n install gcc && \
+    curl -o /usr/local/bin/su-exec.c -SL "https://raw.githubusercontent.com/ncopa/su-exec/master/su-exec.c" && \
+    echo 44d84566f8685273335fc44a82348eb71a6759792c9f0947f7a8be45dbac4d5685e7e6b59cdaf19bd13932f08bb383d9bd4aa37ff56ffac733695e55b5d76391 \
+    /usr/local/bin/su-exec.c | sha512sum -c - && \
```

Comment:
Indent this a bit more.  The way it is it looks like another command rather than part of the previous one.

Reply:
Addressed.